modules/standard/Socket.md

```diff
@@ -0,0 +1,136 @@
+## Socket Module API
+
+To those familiar with the Unix socket API, the method names will feel familiar, though their usage will be somewhat simpler than the raw Unix socket API.
+
+## Connecting to remote sockets
+
+`connect` function will create `TCPConn` instance which will take in the following parameters
+
+**Method:**
+```chapel
+connect(addr:IPAddr,port:int) -> TCPConn
+```
+
+**Parameters:**
+
+```
+addr: IPAddr - ipv4/ipv6 address
+port: Integer - port number
+```
+
+A higher level construct for socket will take in addr as `string` which will we resolved and converted to `IPAddr` internally.
+
+**Method:**
+```chapel
+connect(addr:string,port:int) -> TCPConn
+```
+
+**Parameters:**
+```
+addr: String - host name either in dot-dash or domain form
+port: Integer - port number
+```
+
+The `TCPConn` instance will have methods on it to call
+- `read` ,
+- `write` ,
+- `close`,
+- `closeWrite`,
+- `closeRead`,
+- `setKeepAlive`,
+- `addr` ,
+- `setReadBuffer`
+- and `setWriteBuffer`
+
+The additional support with `TCPConn` and `TCPListener` will be to enable or disable two of the prominent optimization algorithm for TCP Protocol that are:
+
+- `nagle`
+- `quickack`
+
+The support will be in form of separate utility function attached with socket module that can set/unset the optimization based on `boolean` argument
+
+## Creating a Listener
+
+`listen` function will create a `TCPListener` Instance which will take in following parameters
+
+**Method:**
+```chapel
+listen(addr:IPAddr,port:int,backlog=5,reuseaddr=true) -> TCPListener
+```
+
+**Parameters**
+```
+addr: IPAddr - ipv4/ipv6 address
+port: Integer - port number
+backlog: Integer - optional parameter, default value = 0
+reuseaddr: Boolean - optional parameter, default value = true
+```
+
+The `TCPListener` instance will have methods on it to call
+- `accept`,
+- `close`,
+- `addr` and
+- `setSocketOpt` to set various socket options.
+
+`accept` method will return a `TCPConn` instance and associated `addr` of the connection.
```

Comment:
Seems like it would be nice to show `proc` prototypes for each of these methods. In particular for returning the TCPConn and addr, we could use a tuple, or we could use an `out` argument for the addr. But IMO the TCPConn should know the address connected and it would make more sense to just have a methods on the TCPConn - one that return the address of the remote end of the connection and one that return the address of the local end of the connection.

Reply:
Makes sense I think returning `TCPConn` would suffice